rootfs/api/models/build.py

```diff
@@ -78,9 +79,37 @@ def create(self, user, *args, **kwargs):
             raise DeisException(str(e)) from e
 
     def save(self, **kwargs):
-        removed = {}
         previous_release = self.app.release_set.filter(failed=False).latest()
-        if previous_release.build is not None:
+
+        if (
+            settings.DEIS_DEPLOY_REJECT_IF_PROCFILE_MISSING is True and
+            # previous release had a Procfile and the current one does not
+            (
+                previous_release.build is not None and
+                len(previous_release.build.procfile) > 0 and
+                len(self.procfile) == 0
+            )
+        ):
+            # Reject deployment
+            raise Conflict(
+                'Last deployment had a Procfile but is missing in this deploy. '
+                'For a successful deployment provide a Procfile.'
+            )
+
+        # See if processes are permitted to be removed
+        remove_procs = (
+            # If set to True then contents of Procfile does not affect the outcome
+            settings.DEIS_DEPLOY_PROCFILE_MISSING_REMOVE is True or
```

Comment:
@kmala because it's something the user wants, to allow services to push a new build (new image via Jenkins for example) without having to reference the `Procfile` - I see your point tho. See below for improvements.

In any case, I fixed the code and expanded the tests (see api.tests.test_build.BuildTest.test_build_no_remove_process) to prove that running a scale at the end. Prior to my fix that test failed as `build.procfile` was empty. Now it is not and the tests pass.

What are you worried will get corrupt at this point?